File: app-data/src/main/java/com/zxjdev/smile/data/base/BaseDataRepository.java
package com.zxjdev.smile.data.base;

import io.reactivex.ObservableTransformer;
import io.reactivex.android.schedulers.AndroidSchedulers;
import io.reactivex.schedulers.Schedulers;

/**
 * Base class of all the data repositories
 **/
public abstract class BaseDataRepository {

  protected <T> ObservableTransformer<T, T> applyDefaultSchedulerStrategy() {
    return upstream -> upstream.subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread());
  }
}

File: app-data/src/main/java/com/zxjdev/smile/data/base/BaseDataRepository.kt
package com.zxjdev.smile.data.base

import io.reactivex.ObservableTransformer
import io.reactivex.android.schedulers.AndroidSchedulers
import io.reactivex.schedulers.Schedulers

/**
 * Base class of all the data repositories
 */
abstract class BaseDataRepository {

    protected fun <T> applyDefaultSchedulerStrategy(): ObservableTransformer<T, T> {
        return ObservableTransformer { upstream ->
            upstream.subscribeOn(Schedulers.io()).observeOn(AndroidSchedulers.mainThread())
        }
    }
}
